### PR TITLE
Rework LongFormer to make it compatible with ONNX

### PR DESCRIPTION
**Pull request by mfuntowicz**
- [x] Remove implicit `bool` -> `int` conversion while padding attention_mask with `False` value.
- [x] Remove call to `einsum` where `matmul` + `transpose` can be used (makes optimizations easier for ONNX)
- [ ] Diagonal matrix computation try to avoid ScatterND with negative steps / index
- [ ] Use `torch.div(a, b, rounding_mode='trunc')` instead of `floor_divide` 
- [ ] Check [function](https://github.com/huggingface/transformers/blob/master/src/transformers/models/longformer/modeling_longformer.py#L783) reporting `torch.Tensor` return type which doesn't return anything

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
I'm fine with this rework as long as all the slow tests are passing :-)

**Comment by mfuntowicz** [comment]
Hey github actions, I need to find some time to continue my work here ! :D
